app/models/miq_smart_proxy_worker/runner.rb

```diff
@@ -1,3 +1,73 @@
 class MiqSmartProxyWorker::Runner < MiqQueueWorkerBase::Runner
   self.delay_startup_for_vim_broker = true # NOTE: For smartproxy role
+
+  def do_before_work_loop
+    @tid = start_heartbeat_thread
+  end
+
+  def before_exit(message, _exit_code)
+    @exit_requested = true
+    #
+    # Stop the Heartbeat Thread
+    #
+    safe_log("#{message} Stopping Heartbeat Thread.")
+
+    #
+    # Wait for the Heartbeat Thread to stop
+    #
+    unless @tid.nil?
+      safe_log("#{message} Waiting for Heartbeat Thread to Stop.")
+      begin
+        @tid.join(worker_settings[:heartbeat_thread_shutdown_timeout])
+      rescue NoMethodError => join_err
```

Comment:
Under what circumstances would you receive a `NoMethodError`?

Reply:
In the event_catcher, this block is written as

@tid.join(worker_settings[:timeout_value]) rescue nil

Rubocop complained about the rescue clause written that way so I changed it to what was more appropriate - that is if @tid is nil here, you get a NoMethodError on the join (there is no "join" method on a nil).
Now, I realize we just checked for @tid.nil in line 18 above, but my understanding is that the value could be reset between checking and joining.  If that's not correct (in either worker) then we can remove this rescue.